AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
> First, if we're talking non-C++, would heaps even be used?

Yes, with explicit (compiler-specific) address-space annotations on the pointer type.  You'd have to treat it specially and not let these pointers escape "into the wild".  The goal would be to avoid copies by working on this other data more directly.

> Then two modules can't refer to each other's Memories, can they?

You're right, this is the point of the "TODO: JS API for cyclic imports".  Basically, we can support this by a more powerful (but overkill, for the common single-module case) `Instance.instantiate` method that takes multiple modules that can import each other.

> My general point is that, as you said, a GC ref to a buffer is a generalization of this, and we'll have 
> it anyhow, so multiple Memories adds complexity to the spec and to VMs that is not strictly 
> necessary.

I don't think it will, assuming we end up supporting multiple heaps via GC refs in the future; you can think of it as importing a GC value and storing it in an immutable GC global.  If we have the general case, and support multiple imports/exports of the other first-class things, it will seem arbitrary to have this "zero or one" special case with memory forever.